drivers/spi/spi_stm32.c

```diff
@@ -302,43 +378,55 @@ static int spi_dma_move_buffers(const struct device *dev, size_t len)
 static void spi_stm32_send_next_frame(SPI_TypeDef *spi,
 		struct spi_stm32_data *data)
 {
-	const uint8_t frame_size = SPI_WORD_SIZE_GET(data->ctx.config->operation);
+	const uint8_t frame_size = bits2bytes(data->ctx.config->operation);
 	uint32_t tx_frame = SPI_STM32_TX_NOP;
 
-	if (frame_size == 8) {
+	if (frame_size == 1U) {
 		if (spi_context_tx_buf_on(&data->ctx)) {
 			tx_frame = UNALIGNED_GET((uint8_t *)(data->ctx.tx_buf));
 		}
 		LL_SPI_TransmitData8(spi, tx_frame);
-		spi_context_update_tx(&data->ctx, 1, 1);
-	} else {
+	} else if (frame_size == 2U) {
 		if (spi_context_tx_buf_on(&data->ctx)) {
 			tx_frame = UNALIGNED_GET((uint16_t *)(data->ctx.tx_buf));
 		}
 		LL_SPI_TransmitData16(spi, tx_frame);
-		spi_context_update_tx(&data->ctx, 2, 1);
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	} else {
```

Comment:
Nit: SonarQube would prefer (and I think this looks better too)
```suggestion
	} else {
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
```